include/envoy/ratelimit/ratelimit.h

```diff
@@ -67,10 +67,11 @@ class Client {
    * @param domain specifies the rate limit domain.
    * @param descriptors specifies a list of descriptors to query.
    * @param request_id propagates the request_id of the current request to the ratelimit service.
+   * @param span_context propagates current span_context to the ratelimit service.
    */
   virtual void limit(RequestCallbacks& callbacks, const std::string& domain,
-                     const std::vector<Descriptor>& descriptors,
-                     const std::string& request_id) PURE;
+                     const std::vector<Descriptor>& descriptors, const std::string& request_id,
```

Comment:
Instead of adding another param, I think at this point it would be better to make something like:

```
struct TraceContext {
  std::string request_id_;
  std::string span_context_;
};
```

And pass a const TraceContext&

We may add other things in the future and this will end up being a lot cleaner.